lib/ruby_smb/dispatcher/socket.rb

```diff
@@ -30,13 +30,16 @@ def send_packet(packet)
     nil
   end
 
+  # Read a packet off the wire and parse it into a string
+  # Throw Error::NetBiosSessionService if there's an error reading the first 4 bytes,
+  # which are assumed to be the NetBiosSessionService header.
   # @return [String]
   # @todo should return Smb2::Packet
   def recv_packet
     IO.select([@socket])
-    nbss_header = @socket.read(4)
+    nbss_header = @socket.read(4) # Length of NBSS header. TODO: remove to a constant
     if nbss_header.nil?
-      raise "omg"
+      raise RubySmb::Error::NetBiosSessionService
```

Comment:
Can we please include a message with this exception like "NBSS Header missing" or something similar?